src/ng/directive/input.js

```diff
@@ -1431,17 +1431,27 @@ function createDateInputType(type, regexp, parseDate, format) {
   return function dynamicDateInputType(scope, element, attr, ctrl, $sniffer, $browser, $filter) {
     badInputChecker(scope, element, attr, ctrl);
     baseInputType(scope, element, attr, ctrl, $sniffer, $browser);
-    var timezone = ctrl && ctrl.$options.getOption('timezone');
     var previousDate;
+    var previousTimezone;
 
     ctrl.$$parserName = type;
     ctrl.$parsers.push(function(value) {
       if (ctrl.$isEmpty(value)) return null;
+
       if (regexp.test(value)) {
+        var timezone = ctrl.$options.getOption('timezone');
+
         // Note: We cannot read ctrl.$modelValue, as there might be a different
         // parser/formatter in the processing chain so that the model
         // contains some different data format!
+        if (timezone && previousTimezone && previousTimezone !== timezone) {
```

Comment:
Why do we need `timezone` or `previousTimezone` to be truthy? Don't we always need to adjust as long as `previousTimezone !== timezone`?

Reply:
We do need previousTimezone to be defined, since we pass it to addDateMinutes. timezone must not be defined, though.